shared/services/wallet/wallet.go

```diff
@@ -291,6 +292,27 @@ func (w *Wallet) Sign(serializedTx []byte) ([]byte, error) {
 	return signedData, nil
 }
 
+// Signs an arbitrary message using the wallet's private key
+func (w *Wallet) SignMessage(message string) ([]byte, error) {
+	// Get private key
+	privateKey, _, err := w.getNodePrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	fullMessage := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(message), message)
+	hash := crypto.Keccak256Hash([]byte(fullMessage))
+
+	signedMessage, err := crypto.Sign(hash.Bytes(), privateKey)
+	if err != nil {
+		return nil, fmt.Errorf("Error signing message: %w", err)
+	}
+
+	// fix the ECDSA 'v' (see https://bitcoin.stackexchange.com/questions/38351/ecdsa-v-r-s-what-is-v/38909#38909)
+	signedMessage[64] += 27
```

Comment:
lol, wtf

Reply:
Changed the link to point to a better reference. 
Don't blame me, this += 27 comes from the Yellow paper
![image](https://user-images.githubusercontent.com/23104993/183082635-6ad05f1f-64ba-47e7-9870-7e3049576e5b.png)